projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6423,19 +6445,44 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
                 }
             });
         }
+
+        return false;
     }
 
     /**
      * @hidden
      * @internal
      */
-    public endRowEdit(commit = true, event?: Event) {
-        this.endEdit(commit, event);
+    public endRowEdit(commit = true, event?: Event, cancelClick?: boolean) {
+        const canceled = this.endEdit(commit, event);
+
+        /**
+         * If cellEdit was canceled and we are in row edit mode,
+         * we should exit from cell edit if we end the row editing process.
+         */
+        if (this.crudService.cellInEditMode && canceled) {
+            this.crudService.exitCellEdit();
+        }
+
+        if (canceled) {
+            return;
+        }
+
+        /**
+         * If there isn't anything canceled so far,
+         * we should release all editing blocked items.
+         * Because if rowEdit is canceled clicking Done button would block row editing process.
+         */
+        if (!!cancelClick) {
```

Comment:
A cancel should be the same as commit = false (that's the reason it's there I believe)?

There's a weird behavior if you cancel the row edit and try with apply first (which doesn't work as expected) and then cancel - the row edit overlay closes but the call remains oddly marked as changed with it's original value:
![row-edit-cancel](https://user-images.githubusercontent.com/3198469/94137945-f09f0600-fe6f-11ea-916c-1bf2676b0ea9.gif)